bank-accounts.rb

```diff
@@ -51,22 +83,43 @@ def initialize(id, initial_balance, open_date = nil, owner = nil)
         @date = DateTime.strptime(open_date, "%Y-%m-%d %H:%M:%S %z")
       end
       @owner = owner
+      @colors_array = [:red, :light_red, :green, :light_green, :yellow, :light_yellow, :blue, :light_blue, :magenta, :light_magenta, :cyan, :light_cyan]
 
-      raise ArgumentError if @balance < 0
+      raise ArgumentError.new("You cannot open an account with a negative balance.") if @balance < 0
     end
 
     def withdraw(withdrawal)
-      if withdrawal <= @balance
-        @balance -= withdrawal
+      if withdrawal <= 0
+        print "You can only withdraw a positive amount. "
+
+        return @balance
+      end
+
+      if @balance - withdrawal - self.class::FEE >= self.class::MIN_BAL
+        @balance -= (withdrawal + self.class::FEE) if self.class != MoneyMarketAccount
```

Comment:
I like that you used a single-line conditional here but it would be better to use the `if` `else` instead to save you from having the write a similar condition more than once